### DISCUSSION TITLE
Recommendations for persisting chat history locally

### DISCUSSION THREAD

**Discussion by deepinderdeol** [microsoft/semantic-kernel#5860] [Q&A] [answered]
Any recommendations for persisting chat history locally - which database is the best option, Sqlite, DuckDb or something else?

Let's say we keep the most recent 20 messages + the System_Msg in ChatHistory and save the rest to the history-database - most likely the context needed for answering user-queries would be available in ChatHistory, but say it isn't then we could search/retrieve info from the persisted chat-history - does this sound like a reasonable approach or is there a better way to handle these scenarios?

**Comment by sophialagerkranspandey** [comment, marked as answer]
This makes sense as a approach may of the SQL options you described would work.
